src/rust/bitbox02-rust/src/hww/api/restore.rs

```diff
@@ -20,25 +20,42 @@ use pb::response::Response;
 use crate::workflow::{confirm, mnemonic, password, status, unlock};
 
 pub async fn from_file(request: &pb::RestoreBackupRequest) -> Result<Response, Error> {
+    confirm::confirm(&confirm::Params {
+        body: "Restore backup?",
+        accept_is_nextarrow: true,
+        ..Default::default()
+    })
+    .await?;
+
+    let data = match bitbox02::backup::restore_from_directory(&request.id) {
```

Comment:
this seems like a confusing naming. doesn't it simply reads from SD card and optionally tries to recover a corrupt backup? if so a better name is something like "read backup from directory".

also, not sure what the comment at https://github.com/digitalbitbox/bitbox02-firmware/blob/badbe0269/src/backup/restore.c#L245-L247 refers to:

```c
    // if we found a file that matches the checksum, overwrite corrupt files.
    // do majority-vote error correction if all files are corrupt. Read file bit by bit and update
    // the bits that are diverging.
    return RESTORE_OK;
```

Reply:
i meant, i know what it referes to - to the loop above it - but it's just a bit confusing to read the comment below what it's referring to. i was trying to figure out what `restore_from_directory` does exactly.

Reply:
out of scope for this PR.